test/test_index_string.cpp

```diff
@@ -962,56 +957,73 @@ namespace {
 TEST_TYPES(StringIndex_EmbeddedZeroesCombinations, non_nullable, nullable)
 {
     constexpr bool nullable = TEST_TYPE::value;
 
-    // String index
-    ref_type ref = StringColumn::create(Allocator::get_default());
-    StringColumn col(Allocator::get_default(), ref, nullable);
-    const StringIndex& ndx = *col.create_search_index();
-
-    const size_t MAX_LENGTH = 16; // Test medium
-    char tmp[MAX_LENGTH];         // this is a bit of a hack, that relies on the string being copied in column.add()
-
-    for (size_t length = 1; length <= MAX_LENGTH; ++length) {
+#if TEST_DURATION == 0
+    for (int seed = 0; seed < 5; seed++) {
+#else 
+    for (int seed = 0; seed < 100; seed++) {
+#endif
+        // String index
+        ref_type ref = StringColumn::create(Allocator::get_default());
+        StringColumn col(Allocator::get_default(), ref, nullable);
+        const StringIndex& ndx = *col.create_search_index();
 
-        {
-            Random random(42);
-            const size_t combinations = 1 << length;
-            for (size_t i = 0; i < combinations; ++i) {
-                StringData str = create_string_with_nuls(i, length, tmp, random);
-                col.add(str);
+#if TEST_DURATION == 0
+        const size_t MAX_LENGTH = 8;
+#else
+        const size_t MAX_LENGTH = 16; // Test medium length strings
+#endif
+        char tmp[MAX_LENGTH];
+
+        for (size_t length = 1; length <= MAX_LENGTH; ++length) {
+            {
+                Random random(seed);
+                const size_t combinations = 1 << length;
+                for (size_t i = 0; i < combinations; ++i) {
+                    StringData str = create_string_with_nuls(i, length, tmp, random);
+                    col.add(str);
+                }
             }
-        }
 
-        // check index up to this length
-        size_t expected_index = 0;
-        for (size_t l = 1; l <= length; ++l) {
-            Random random(42);
-            const size_t combinations = 1 << l;
-            for (size_t i = 0; i < combinations; ++i) {
-                StringData needle = create_string_with_nuls(i, l, tmp, random);
-                CHECK_EQUAL(ndx.find_first(needle), expected_index);
-                CHECK(strncmp(col.get(expected_index).data(), needle.data(), l) == 0);
-                CHECK_EQUAL(col.get(expected_index).size(), needle.size());
-                expected_index++;
+            // check index up to this length
+            size_t expected_index = 0;
+            for (size_t l = 1; l <= length; ++l) {
+                Random random(seed);
+                const size_t combinations = 1 << l;
+                for (size_t i = 0; i < combinations; ++i) {
+                    StringData needle = create_string_with_nuls(i, l, tmp, random);
+                    CHECK_EQUAL(ndx.find_first(needle), expected_index);
+                    CHECK(strncmp(col.get(expected_index).data(), needle.data(), l) == 0);
```

Comment:
If you could merge this check and the check below into one single check by using the `StringData operator==` implicitly in `CHECK(col.get(expected_index), needle)` because that would check the size and the data in one call. Might speed things up a bit too.